core/storage/sqlite3_ondisk.rs

```diff
@@ -526,12 +556,7 @@ impl PageContent {
         let ncells = self.cell_count();
         // the page header is 12 bytes for interior pages, 8 bytes for leaf pages
         // this is because the 4 last bytes in the interior page's header are used for the rightmost pointer.
-        let cell_pointer_array_start = match self.page_type() {
```

Comment:
is this a bug? doesn't include `offset` for rootpage

Reply:
hmmm feels like it.

Reply:
actually this one in particular probably isn't, because `PageContent` generally tries to hide the existence of the potential 100 byte offset from the user -- all of the read/write functions e.g. `read_u16()` add the offset under the hood:

```
    pub fn read_u16(&self, pos: usize) -> u16 {
        let buf = self.as_ptr();
        u16::from_be_bytes([buf[self.offset + pos], buf[self.offset + pos + 1]])
    }
```

whereas this quoted code does the following:

```
let cell_pointer_array_start = (...does NOT include offset...)
let cell_pointer = cell_pointer_array_start + (idx * 2);
let cell_pointer = self.read_u16(cell_pointer) as usize; // <--key part, offset added implicitly by read_u16
```

so the offset actually MUST not be added here because then it would be double-added :)

there might be a way to refactor this to make it less footgun-y

Reply:
And I'm blindly guessing that your thought process when naming `pub fn cell_get_raw_pointer_region()` was that 'raw' means that it includes the offset

Reply:
I'm still thinking there is a 100-byte offset related error _somewhere_ that is causing CREATE TABLE errors

Reply:
I think we should be very intentional about commenting/variable naming when we are dealing with values that do or do not include `self.offset` (which is always 0 except for the rootpage for which it is 100 due to the db header), because otherwise it gets very confusing. i don't have a great solution for this yet

Reply:
> I think we should be very intentional about commenting/variable naming when we are dealing with values that do or do not include self.offset (which is always 0 except for the rootpage for which it is 100 due to the db header), because otherwise it gets very confusing. i don't have a great solution for this yet

I'm okay with that, I feel like we should strive to never use offset but there are cases where it's necessary.